### PR TITLE
DOCSP-33414 Add section on FTS to RQL reference

### PR DESCRIPTION
**Pull request by lindseymoore**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-33414

- [Realm Query Language](https://preview-mongodblindseymoore.gatsbyjs.io/realm/DOCSP-33414/realm-query-language/#full-text-search): Update overall code examples with FTS annotation on `Item.name` property. Add section on FTS. Behind the scenes, started a new RQL test file to reflect new JSv12 standards. 

### Reminder Checklist

Before merging your PR, make sure to check a few things.

- [ ] Did you tag pages appropriately?
  - genre
  - meta.keywords
  - meta.description
- [x] Describe your PR's changes in the Release Notes section
- [x] Create a Jira ticket for related docs-app-services work, if any

### Release Notes

**Other**
- Realm Query Language: Update RQL reference with information on querying with full-text search.
### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: examples/node/v12/__tests__/realm-query-language.test.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      );
+
+    const itemsStartingWithWri = items.filtered(
+        // :remove-end:
+        // Find items starting with 'wri-' using '*'
+        "name TEXT $0", "wri*"
+        // :snippet-end:
+      );
+
+    expect(itemsWithWrite.length).toBeGreaterThan(0);
```
Comment (cbullinger) [inline examples/node/v12/__tests__/realm-query-language.test.js:132]: do we want to make these actual values more specific to ensure the search syntax is working?

File: examples/dart/test/task_project_models_test.dart [hunk 2]
```diff
@@ -12,7 +12,7 @@ class _Item {
   @MapTo("_id")
   @PrimaryKey()
   late ObjectId id;
-
+  @Indexed(RealmIndexType.fullText)
```
Comment (cbullinger) [inline examples/dart/test/task_project_models_test.dart:15]: doesn't this model need to be re-generated with the change?

**Review by cbullinger — CHANGES_REQUESTED** [review]
Some corrections needed

File: source/realm-query-language.txt [hunk 3]
```diff
@@ -905,7 +943,7 @@ To query geospatial data:
 
 1. Create an object with a property containing the embedded geospatial data.
 2. Define the geospatial shape to set the boundary for the query.
-3. Query using the GEOWITHIN RQL operator. 
+3. Query using the ``geoWithin`` RQL operator.
```
[the remark this answers is not in the document]
Comment (lindseymoore) [reply source/realm-query-language.txt:946]: According to this design doc, seems to be camel case? https://docs.google.com/document/d/1Z7CK79CqebEtVllVHTNEq8neKhoMWeFCDjKBrn_pq1Y/edit#heading=h.2s9zdv6a57gn
I'll update

File: source/realm-query-language.txt [hunk 4]
```diff
@@ -884,6 +885,43 @@ operators interact with lists and comparison operators:
...
+.. _rql-fts:
+
+Full Text Search
+----------------
+
+You can use RQL to query on properties that have a full-text search (FTS)
+annotation. For information on enabling FTS on a property, see the FTS 
+documentation for your SDK:
+
+- :ref:`Flutter SDK <flutter-fts-annotation>`
```
[the remark this answers is not in the document]
Comment (lindseymoore) [reply source/realm-query-language.txt:897]: Not sure it's necessary, this page doesn't include any information that those pages won't have. Is it more to promote this page?

File: source/realm-query-language.txt [hunk 5]
```diff
@@ -919,8 +961,8 @@ see the geospatial documentation for your SDK:
 - :ref:`Flutter SDK <flutter-data-types-geospatial>`
 - :ref:`Kotlin SDK <kotlin-geospatial>`
```
Comment (cbullinger) [inline source/realm-query-language.txt:962]: I updated the Read page with info on querying geo data. it's probably a more appropriate link here ... but I'm realizing that I didn't add a ref anchor to the section so nm 🤷

**Review by cbullinger — APPROVED** [review]
a few suggestions, but lgtm! Also, should we plan to eventually get all of the RQL examples in the new v12 file(s)?

File: source/realm-query-language.txt [hunk 6]
```diff
@@ -884,6 +885,47 @@ operators interact with lists and comparison operators:
...
+.. literalinclude:: /examples/generated/realm-query-language/realm-query-language.test.snippet.rql-fts.js
+     :language: js
+
+Full-Text Search Tokenizer Details
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Full-Text Search (FTS) indexes support:
+
+- Tokens are diacritics- and case-insensitive.
+- Tokens can only consist of characters from ASCII and the Latin-1 supplement (western languages).
```
[the remark this answers is not in the document]
Comment (lindseymoore) [reply source/realm-query-language.txt:924]: Missed this one, thanks!

File: source/realm-query-language.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (lindseymoore) [reply source/realm-query-language.txt:950]: Created!: https://jira.mongodb.org/browse/DOCSP-37318

**Pull request merged**